### ISSUE TITLE
MSE Loss fails to converge during Stage 2 training

### ISSUE THREAD

**Issue by lyfiter** [Trae1ounG/DyPRAG#21]
Thank you for this impressive work on DyPRAG. I am currently trying to reproduce the Stage 2 (DyPRAG Training) process, but I encountered a significant issue where the MSE Loss fails to decrease, indicating that the ParameterTranslator (Projector) is unable to learn the mapping from input embeddings to LoRA weights.

1. Issue Description
When running dyprag_train.py, I observed that:
The LM Loss (Language Modeling loss) drops quickly (e.g., from 2.5 to 1.2), likely because the model overfits the current batch via the injected parameters.
However, the MSE Loss (alignment between predicted parameters and Stage 1 Ground Truth LoRA) remains high or fluctuates without a clear downward trend.
2. Attempts to Fix
Suspecting that the optimization might be dominated by the LM Loss, I tried a "Warm-up" strategy by setting the weights of lm_loss and kl_loss to 0, focusing purely on the MSE Loss

3. My Hypotheses & Questions
I suspect the issue might be related to the architecture or data distribution:
Input/Output Distribution Gap: The input input_embeds (hidden states from the LLM) and the target lora_state_dict (LoRA weights) have vastly different scales and distributions.
Question: Did you apply any specific Normalization (e.g., LayerNorm) or Initialization tricks to the ParameterTranslator or Projector class to handle this?
Model Capacity: The current Projector is a simple MLP with ReLU.
Question: Is this simple architecture sufficient to regress high-dimensional LoRA matrices? Or is there a specific hyperparameter (like projector_p) that is critical for convergence?
Data Quality:
Question: How sensitive is Stage 2 to the quality of Stage 1 LoRA weights? If Stage 1 weights are too diverse or noisy, would it prevent the projector from converging?
Reproduce Config:
python3 -u src/train_dyprag.py \
    --model_name=llama3.2-1b-instruct \
    --datasets=2wikimultihopqa \
    --learning_rate=0.0003 \
    --lora_rank=2 \
    --lora_alpha=32 \
    --max_new_tokens=128 \
    --sample_rate=0.1 \
    --dyprag_learning_rate=1e-5 \
    --dyprag_train_epochs=1

**Comment by Trae1ounG** [comment]
Thanks for your interest. 

1.The mse is an auxiliary loss that help parameter translator to reconstruct P, but P may not be the best because the lm loss and kl loss is more important for alignment. So the loss of mse is more likely to vibrate.

2.Only mse will not work, since it is hard to only learn the consistent values.

3.1/3.2 Input/Output Distribution Gap: That's a verying important point of hypernetwork. Honestly we didn't try different architecture. We think the idea of this process of online document parametrization is more important. But it's worth to find what architeture is more fit for this cross modality transformation.

3.3: How sensitive is Stage 2 to the quality of Stage 1 LoRA weights? Accually we add this experiment in our newly submitted version, that the quality of stage 1 lora weights indeed influce the performance of stage 2. 

<img width="1242" height="552" alt="Image" src="https://github.com/user-attachments/assets/98d8306a-6f63-4b23-a278-638ef9eb8142" />
